### PR TITLE
Add option for filtering only directly-requested PRs

### PR DESCRIPTION
**Pull request by teejae**
- Add checkbox for filtering only directly-requested PRs
- Allow including list of whitelisted team names

Screenshot:
Default Off:
image

On, with whitelisted team names:
image

### CODE REVIEW

**Comment by teejae** [comment]
@fwouts, added some tests, per offline request.

File: src/components/PullRequestList.tsx [hunk 1]
```diff
@@ -38,47 +61,107 @@ export interface PullRequestListProps {
   emptyMessage: string;
   mutingConfiguration: "allow-muting" | "allow-unmuting" | "none";
   newCommitsNotificationToggled: boolean | null;
+  onlyDirectRequestsToggled: boolean | null;
```
Comment (fwouts) [inline src/components/PullRequestList.tsx:64]: This is adding quite a few props to `PullRequestListProps` for a smaller part of the UI. I think we may want to extract it all into a new component, then pass it as a single prop of type `ReactNode` here. This pattern would also be useful for the existing `newCommitsNotificationToggled` prop.

I'm happy to merge your PR as-is and make the change myself, unless you specifically feel like taking it on.

File: src/filtering/filters.spec.ts [hunk 2]
```diff
@@ -50,6 +60,61 @@ describe("filters (incoming)", () => {
...
+          .seenAs("kevin")
+          .teams({
+            team: ["kevin"],
+          })
+          .reviewRequested([], ["team"])
+          .build()
+      )
+    ).toEqual([Filter.INCOMING]);
+  });
+  it("is INCOMING when the user is in a reviewer team, but only wants whitelisted teams", () => {
```
Comment (fwouts) [inline src/filtering/filters.spec.ts:81]:
```suggestion
  it("is NOTHING when the user is in a reviewer team, but only wants whitelisted teams", () => {
```

File: src/filtering/filters.spec.ts [hunk 3]
```diff
@@ -195,6 +260,56 @@ describe("filters (incoming)", () => {
...
+          .author("fwouts")
+          .seenAs("kevin")
+          .reviewRequested(["kevin"])
+          // Another user posted a review after we muted.
+          .addReview("dries", "CHANGES_REQUESTED", 200)
+          .build()
+      )
+    ).toEqual([Filter.MUTED]);
+  });
   it("is MUTED when the PR is muted until next update and the author did not add new comments or reviews", () => {
```
Comment (fwouts) [inline src/filtering/filters.spec.ts:313]: This test currently has the same description as the one above, but it's a different scenario (when a team is requested). Please rename.

**Review by fwouts — APPROVED** [review]
Awesome work!

File: src/filtering/filters.spec.ts (hunk 3 above)
Comment (teejae) [reply src/filtering/filters.spec.ts]: done

File: src/filtering/filters.spec.ts (hunk 2 above)
Comment (teejae) [reply src/filtering/filters.spec.ts]: done

File: src/components/PullRequestList.tsx (hunk 1 above)
Comment (teejae) [reply src/components/PullRequestList.tsx:64]: would you be able to do the refactor? feel free to do it in this PR directly before merge.

**Review by teejae — COMMENTED** [review]
@fwouts FYI, I've changed the button over to the themed button, adding support for disabling.

**Pull request merged**